[bug] errors with  closeButtonProps in dmc.Modal (and others ...ButtonProps).  Issues with clear button and clearButtonProps

Hi,

Since 0.15.2 my Modal Is creating a React error.

It happens when i'm using closeButtonProps={"children" : dmc.Text() } (or any Components, dash.html included). 

Sample to reproduce bug :

```python
import dash_mantine_components as dmc
from dash import Dash, _dash_renderer, html
import datetime

_dash_renderer._set_react_version("18.2.0")

app = Dash(__name__)

app.layout = dmc.MantineProvider(
    dmc.Center(
        [
            dmc.Modal(
                title="New Modal",
                id="modal-simple",
                opened=True,
                children=[
                    dmc.Text("I am a modal component."),
                ],
                closeButtonProps={
                    "children" : html.H1("Hi")
                }
            ),
            # dmc.DatePickerInput(
            #     value=datetime.datetime.now().date(),
            #     label="Date clearable",
            #     w=200,
            #     clearable=True,
            #     clearButtonProps={
            #         "children" : dmc.Text("Hi")
            #     }
            # ),
        ],
        style={"height": '100vh', "width": "100vw"},
    )
)



if __name__ == "__main__":
    app.run_server(debug=True)
```

Error from Console : 

```console
hook.js:608 Error: Objects are not valid as a React child (found: object with keys {props, type, namespace}). If you meant to render a collection of children, use an array instead.
    at throwOnInvalidObjectType (react-dom@18.v2_17_1m1721817131.2.0.js:14897:11)
    at createChild (react-dom@18.v2_17_1m1721817131.2.0.js:15149:9)
    at reconcileChildrenArray (react-dom@18.v2_17_1m1721817131.2.0.js:15414:27)
    at reconcileChildFibers (react-dom@18.v2_17_1m1721817131.2.0.js:15831:18)
    at reconcileChildren (react-dom@18.v2_17_1m1721817131.2.0.js:19177:30)
    at updateHostComponent (react-dom@18.v2_17_1m1721817131.2.0.js:19934:5)
    at beginWork (react-dom@18.v2_17_1m1721817131.2.0.js:21628:16)
    at beginWork$1 (react-dom@18.v2_17_1m1721817131.2.0.js:27436:16)
    at performUnitOfWork (react-dom@18.v2_17_1m1721817131.2.0.js:26567:14)
    at workLoopSync (react-dom@18.v2_17_1m1721817131.2.0.js:26476:7) Error Component Stack:
    at ComponentErrorBoundary (ComponentErrorBoundary.react.js:6:24)
    at BaseTreeContainer (TreeContainer.js:101:24)
    at DashWrapper (TreeContainer.js:88:31)
    at div (<anonymous>)
    at dash_mantine_components.v0_15_2m1737992795.js:2:1641474
    at dash_mantine_components.v0_15_2m1737992795.js:2:1314963
    at l (dash_mantine_components.v0_15_2m1737992795.js:2:1021928)
    at CheckedComponent (TreeContainer.js:46:12)
    at ComponentErrorBoundary (ComponentErrorBoundary.react.js:6:24)
    at BaseTreeContainer (TreeContainer.js:101:24)
    at DashWrapper (TreeContainer.js:88:31)
    at c (dash_mantine_components.v0_15_2m1737992795.js:2:1664578)
    at p (dash_mantine_components.v0_15_2m1737992795.js:2:1662861)
    at l (dash_mantine_components.v0_15_2m1737992795.js:2:1190940)
    at CheckedComponent (TreeContainer.js:46:12)
    at ComponentErrorBoundary (ComponentErrorBoundary.react.js:6:24)
    at BaseTreeContainer (TreeContainer.js:101:24)
    at DashWrapper (TreeContainer.js:88:31)
    at div (<anonymous>)
    at div (<anonymous>)
    at div (<anonymous>)
    at GlobalErrorOverlay (GlobalErrorOverlay.react.js:9:24)
    at div (<anonymous>)
    at DebugMenu (DebugMenu.react.js:45:24)
    at div (<anonymous>)
    at UnconnectedGlobalErrorContainer (GlobalErrorContainer.react.js:7:24)
    at ConnectFunction (connect.js:218:75)
    at UnconnectedContainer (APIController.react.js:35:9)
    at ConnectFunction (connect.js:218:75)
    at UnconnectedAppContainer (AppContainer.react.js:13:24)
    at ConnectFunction (connect.js:218:75)
    at Provider (Provider.js:7:3)
    at AppProvider (AppProvider.react.tsx:6:24)
```

Same problem with DatePickerInput's clearButtonProps property, so i'm guessing it's impacting every ...ButtonProps

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Change Log
 
+# Unreleased
+
+### Fixed
+
+- enable components to be used in `children` and `icon` in the `closeButtonProps` and `clearButtonProps`. #493 by @AnnMarieW
+
 # 0.15.2
 
 ### Added

--- a/src/ts/components/core/combobox/MultiSelect.tsx
+++ b/src/ts/components/core/combobox/MultiSelect.tsx
@@ -3,6 +3,7 @@ import { useDebouncedValue, useDidUpdate, useFocusWithin } from "@mantine/hooks"
 import { BoxProps } from "props/box";
 import { ComboboxLikeProps } from "props/combobox";
 import { DashBaseProps, PersistenceProps, DebounceProps } from "props/dash";
+import { __ClearButtonProps } from "props/button";
 import { __BaseInputProps } from "props/input";
 import { ScrollAreaProps } from "props/scrollarea";
 import { StylesApiProps } from "props/styles";
@@ -36,7 +37,7 @@ interface Props
     /** Determines whether the clear button should be displayed in the right section when the component has value, `false` by default */
     clearable?: boolean;
     /** Props passed down to the clear button */
-    clearButtonProps?: object;
+    clearButtonProps?: __ClearButtonProps;
     /** Props passed down to the hidden input */
     hiddenInputProps?: object;
     /** Divider used to separate values in the hidden input `value` attribute, `','` by default */

--- a/src/ts/components/core/combobox/Select.tsx
+++ b/src/ts/components/core/combobox/Select.tsx
@@ -3,6 +3,7 @@ import { useDebouncedValue, useDidUpdate } from "@mantine/hooks";
 import { BoxProps } from "props/box";
 import { ComboboxLikeProps } from "props/combobox";
 import { DashBaseProps, PersistenceProps, DebounceProps } from "props/dash";
+import { __ClearButtonProps } from "props/button";
 import { __BaseInputProps } from "props/input";
 import { ScrollAreaProps } from "props/scrollarea";
 import { StylesApiProps } from "props/styles";
@@ -34,7 +35,7 @@ interface Props
     /** Determines whether the clear button should be displayed in the right section when the component has value, `false` by default */
     clearable?: boolean;
     /** Props passed down to the clear button */
-    clearButtonProps?: object;
+    clearButtonProps?: __ClearButtonProps;
     /** Props passed down to the hidden input */
     hiddenInputProps?: object;
     /** Props passed down to the underlying `ScrollArea` component in the dropdown */

--- a/src/ts/components/core/combobox/TagsInput.tsx
+++ b/src/ts/components/core/combobox/TagsInput.tsx
@@ -6,6 +6,7 @@ import { useDidUpdate } from "@mantine/hooks";
 import { BoxProps } from "props/box";
 import { ComboboxLikeProps } from "props/combobox";
 import { DashBaseProps, PersistenceProps } from "props/dash";
+import { __ClearButtonProps } from "props/button";
 import { __BaseInputProps } from "props/input";
 import { ScrollAreaProps } from "props/scrollarea";
 import { StylesApiProps } from "props/styles";
@@ -34,7 +35,7 @@ interface Props
     /** Determines whether the clear button should be displayed in the right section when the component has value, `false` by default */
     clearable?: boolean;
     /** Props passed down to the clear button */
-    clearButtonProps?: object;
+    clearButtonProps?: __ClearButtonProps;
     /** Props passed down to the hidden input */
     hiddenInputProps?: object;
     /** Divider used to separate values in the hidden input `value` attribute, `','` by default */

--- a/src/ts/components/dates/DateInput.tsx
+++ b/src/ts/components/dates/DateInput.tsx
@@ -8,6 +8,7 @@ import {
     MonthLevelSettings,
     YearLevelSettings,
 } from "props/dates";
+import { __ClearButtonProps } from "props/button";
 import { __BaseInputProps } from "props/input";
 import { PopoverProps } from "props/popover";
 import { StylesApiProps } from "props/styles";
@@ -35,7 +36,7 @@ interface Props
     /** Determines whether input value can be cleared, adds clear button to right section, false by default */
     clearable?: boolean;
     /** Props added to clear button */
-    clearButtonProps?: object;
+    clearButtonProps?: __ClearButtonProps;
     /** Dayjs format to display input value, "MMMM D, YYYY" by default  */
     valueFormat?: string;
     /** Determines whether input value should be reverted to last known valid value on blur, true by default */

--- a/src/ts/props/button.ts
+++ b/src/ts/props/button.ts
@@ -1,0 +1,17 @@
+import { MantineRadius, MantineSize } from "@mantine/core";
+import React from "react";
+
+export interface __CloseButtonProps {
+    /** Controls width and height of the button. Numbers are converted to rem. `'md'` by default. */
+    size?: MantineSize | (string & {}) | number;
+    /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
+    radius?: MantineRadius;
+    /** Sets `disabled` and `data-disabled` attributes on the button element */
+    disabled?: boolean;
+    /** `X` icon `width` and `height`, `80%` by default */
+    iconSize?: number | string;
+    /** Content rendered inside the button, for example `VisuallyHidden` with label for screen readers */
+    children?: React.ReactNode;
+    /** Replaces default close icon. If set, `iconSize` prop is ignored. */
+    icon?: React.ReactNode;
+}

--- a/src/ts/props/button.ts
+++ b/src/ts/props/button.ts
@@ -1,10 +1,8 @@
 import { MantineRadius, MantineSize } from "@mantine/core";
 import React from "react";
 
-export interface __CloseButtonProps {
-    /** Controls width and height of the button. Numbers are converted to rem. `'md'` by default. */
-    size?: MantineSize | (string & {}) | number;
-    /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
+export interface __BaseButtonProps {
+     /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
     radius?: MantineRadius;
     /** Sets `disabled` and `data-disabled` attributes on the button element */
     disabled?: boolean;
@@ -16,18 +14,12 @@ export interface __CloseButtonProps {
     icon?: React.ReactNode;
 }
 
+export interface __CloseButtonProps extends __BaseButtonProps {
+    /** Controls width and height of the button. Numbers are converted to rem. `'md'` by default. */
+    size?: MantineSize | (string & {}) | number;
+}
 
-export interface __ClearButtonProps {
+export interface __ClearButtonProps extends __BaseButtonProps {
     /** Size of the button, by default value is based on input context */
     size?: MantineSize | (string & {});
-    /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
-    radius?: MantineRadius;
-    /** Sets `disabled` and `data-disabled` attributes on the button element */
-    disabled?: boolean;
-    /** `X` icon `width` and `height`, `80%` by default */
-    iconSize?: number | string;
-    /** Content rendered inside the button, for example `VisuallyHidden` with label for screen readers */
-    children?: React.ReactNode;
-    /** Replaces default close icon. If set, `iconSize` prop is ignored. */
-    icon?: React.ReactNode;
 }

--- a/src/ts/props/button.ts
+++ b/src/ts/props/button.ts
@@ -15,3 +15,19 @@ export interface __CloseButtonProps {
     /** Replaces default close icon. If set, `iconSize` prop is ignored. */
     icon?: React.ReactNode;
 }
+
+
+export interface __ClearButtonProps {
+    /** Size of the button, by default value is based on input context */
+    size?: MantineSize | (string & {});
+    /** Key of `theme.radius` or any valid CSS value to set border-radius. Numbers are converted to rem. `theme.defaultRadius` by default. */
+    radius?: MantineRadius;
+    /** Sets `disabled` and `data-disabled` attributes on the button element */
+    disabled?: boolean;
+    /** `X` icon `width` and `height`, `80%` by default */
+    iconSize?: number | string;
+    /** Content rendered inside the button, for example `VisuallyHidden` with label for screen readers */
+    children?: React.ReactNode;
+    /** Replaces default close icon. If set, `iconSize` prop is ignored. */
+    icon?: React.ReactNode;
+}

--- a/src/ts/props/dates.ts
+++ b/src/ts/props/dates.ts
@@ -7,6 +7,7 @@ import {
 } from "@mantine/dates";
 import { BoxProps } from "./box";
 import { __BaseInputProps } from "./input";
+import { __ClearButtonProps } from "./button";
 import { ModalProps } from "./modal";
 import { PopoverProps } from "./popover";
 import { StylesApiProps } from "./styles";
@@ -126,7 +127,7 @@ export interface DateInputSharedProps extends Omit<__BaseInputProps, "size"> {
     /** Determines whether input value can be cleared, adds clear button to right section, false by default */
     clearable?: boolean;
     /** Props passed down to clear button */
-    clearButtonProps?: object;
+    clearButtonProps?: __ClearButtonProps;
     /** Determines whether the user can modify the value */
     readOnly?: boolean;
     /** Determines whether dates value should be sorted before onChange call, only applicable when type="multiple", true by default */

--- a/src/ts/props/modal.ts
+++ b/src/ts/props/modal.ts
@@ -5,8 +5,10 @@ import {
     MantineSpacing,
 } from "@mantine/core";
 import { BoxProps } from "./box";
+import { __CloseButtonProps } from "./button";
 import { OverlayProps } from "./overlay";
 import { TransitionProps } from "./transition";
+
 
 export interface ModalBaseProps extends BoxProps {
     /** If set modal/drawer will not be unmounted from the DOM when it is hidden, `display: none` styles will be added instead, `false` by default */
@@ -50,6 +52,10 @@ export interface ModalBaseOverlayProps
 }
 
 
+export interface ModalBaseCloseButtonProps
+    extends __CloseButtonProps,
+        BoxProps {}
+
 export interface ModalProps extends ModalBaseProps {
     /** Modal title */
     title?: React.ReactNode;
@@ -62,7 +68,7 @@ export interface ModalProps extends ModalBaseProps {
     /** Determines whether the close button should be rendered, `true` by default */
     withCloseButton?: boolean;
     /** Props passed down to the close button */
-    closeButtonProps?: object;
+    closeButtonProps?: ModalBaseCloseButtonProps;
     /** Top/bottom modal offset, `5dvh` by default */
     yOffset?: React.CSSProperties["marginTop"];
     /** Left/right modal offset, `5vw` by default */

--- a/tests/dates/test_date_input.py
+++ b/tests/dates/test_date_input.py
@@ -24,6 +24,17 @@ debounce = dmc.Stack(
             debounce=2000,
         ),
         dmc.Box(id="out-2000"),
+
+        # ensure clearButton children and icon can render with components without error
+        dmc.DateInput(
+            id="date-input-clearable",
+            value="2025-01-01",
+            clearable=True,
+            clearButtonProps = {
+                "children": html.Div("x", id="id1"),
+                "icon": html.Div("y",  id="id2"),
+            }
+        )
     ]
 )
 

--- a/tests/dates/test_date_picker.py
+++ b/tests/dates/test_date_picker.py
@@ -40,6 +40,16 @@ def make_app(**kwargs):
                 maxDate="2024-10-31",
                 **kwargs
             ),
+            # ensure clearButton children and icon can render with components without error
+            dmc.DatePickerInput(
+                id="date-picker-input-clearable",
+                value="2025-01-01",
+                clearable=True,
+                clearButtonProps={
+                    "children": html.Div("x", id="id1"),
+                    "icon": html.Div("y", id="id2"),
+                }
+            )
         ]
     )
 

--- a/tests/test_modal.py
+++ b/tests/test_modal.py
@@ -1,0 +1,50 @@
+from dash import Dash, html, Output, Input, State, _dash_renderer
+import dash_mantine_components as dmc
+
+_dash_renderer._set_react_version("18.2.0")
+
+
+def test_001mo_modal(dash_duo):
+    app = Dash(__name__)
+
+    component = html.Div(
+        [
+            dmc.Button("Open Modal", id="modal-demo-button"),
+            dmc.Modal(
+                title="New Modal",
+                id="modal-simple",
+                closeButtonProps={
+                    "children": html.Div(" close", id="my-component"),
+                    "icon": html.Div("x", id="modal-custom-close"),
+                },
+                children=[
+                    dmc.Text("I am in a modal component."),
+                ],
+            ),
+        ]
+    )
+
+    app.layout = dmc.MantineProvider(
+       component
+    )
+
+    @app.callback(
+        Output("modal-simple", "opened"),
+        Input("modal-demo-button", "n_clicks"),
+        State("modal-simple", "opened"),
+        prevent_initial_call=True,
+    )
+    def modal_demo(_, opened):
+        return not opened
+
+    dash_duo.start_server(app)
+
+    # Open modal
+    modal_btn = dash_duo.find_element("#modal-demo-button")
+    modal_btn.click()
+
+    # Close modal
+    modal_close_btn = dash_duo.find_element("#modal-custom-close")
+    modal_close_btn.click()
+
+    assert dash_duo.get_logs() == []

